diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/GcTimeMonitor.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/GcTimeMonitor.java
index 95d0d4d290c..4c14647dea4 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/GcTimeMonitor.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/GcTimeMonitor.java
@@ -138,7 +138,8 @@ public GcTimeMonitor(long observationWindowMs, long sleepIntervalMs,
     bufSize = (int) (observationWindowMs / sleepIntervalMs + 2);
     // Prevent the user from accidentally creating an abnormally big buffer,
     // which will result in slow calculations and likely inaccuracy.
-    Preconditions.checkArgument(bufSize <= 128 * 1024);
+    Preconditions.checkArgument(
+            bufSize <= 128 * 1024 && bufSize > 0);
     gcDataBuf = new TsAndData[bufSize];
     for (int i = 0; i < bufSize; i++) {
       gcDataBuf[i] = new TsAndData();
